<?php
namespace App\Exceptions; use App\Library\Response; use Exception; use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler; use Illuminate\Support\Facades\Log; class Handler extends ExceptionHandler { protected $dontReport = array(); protected $dontFlash = array('password', 'password_confirmation'); public function report(Exception $sp579fb8) { parent::report($sp579fb8); } private function msg($sp85ba11, $sp469aa0, $sp7a1202) { return $sp85ba11->isXmlHttpRequest() ? response()->json(array('message' => $sp469aa0), $sp7a1202) : response()->view('errors._', array('code' => $sp7a1202, 'message' => $sp469aa0), $sp7a1202); } public function render($sp85ba11, Exception $sp1b3a33) { if ($sp1b3a33 instanceof \Illuminate\Database\Eloquent\ModelNotFoundException) { return $this->msg($sp85ba11, '记录未找到', 404); } elseif ($sp1b3a33 instanceof \Illuminate\Auth\AuthenticationException) { return $this->msg($sp85ba11, '您需要登录您的账户再进行此操作', 401); } elseif ($sp1b3a33 instanceof \Illuminate\Auth\Access\AuthorizationException) { return $this->msg($sp85ba11, '未授权的操作', 403); } elseif ($sp1b3a33 instanceof \Illuminate\Validation\ValidationException) { return parent::render($sp85ba11, $sp1b3a33); } elseif ($sp1b3a33 instanceof \Illuminate\Session\TokenMismatchException) { return $this->msg($sp85ba11, '请刷新页面后重试', 403); } elseif ($sp1b3a33 instanceof \Symfony\Component\HttpKernel\Exception\NotFoundHttpException) { return $this->msg($sp85ba11, '页面未找到', 404); } elseif ($sp1b3a33 instanceof \Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException) { return $this->msg($sp85ba11, '请求方法不允许', 405); } elseif ($sp1b3a33 instanceof \Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException) { Log::error('Caught a ServiceUnavailableHttpException', array('Exception' => $sp1b3a33)); return $this->msg($sp85ba11, '当前服务不可用，请稍后重试', 503); } elseif ($sp1b3a33 instanceof \Symfony\Component\HttpKernel\Exception\HttpException) { $sp201fea = $sp85ba11->isXmlHttpRequest(); switch ($sp201fea) { case 429: return $this->msg($sp85ba11, '您的请求过于频繁，请稍后重试', $sp201fea); break; default: Log::error('Caught a UnknownHttpException', array('Exception' => $sp1b3a33)); return $this->msg($sp85ba11, '当前服务不可用，请稍后重试', $sp201fea); } } Log::error('Uncaught Exception', array('Exception' => $sp1b3a33)); if (config('app.debug')) { return parent::render($sp85ba11, $sp1b3a33); } else { return $this->msg($sp85ba11, '未知错误，请查看错误日志（storage/logs）', 500); } } }